### PR TITLE
AdaBoost binding tests

### PR DESCRIPTION
**Pull request by nikhilgoel1997**
Contains binding test for adaboost algorithm as asked in issue #1152

### CODE REVIEW

**Comment by nikhilgoel1997** [comment]
One of the 3 builds of travis terminated without starting and gave this in the log - 

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated

Is there something wrong from my end?

File: src/mlpack/tests/main_tests/adaboost_test.cpp [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+  mlpackMain();
+
+  const string weakLearner = CLI::GetParam<string>("weak_learner");
+  if (weakLearner == "perceptron")
+  {
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+  }
```
Comment (rcurtin) [inline src/mlpack/tests/main_tests/adaboost_test.cpp]: I don't understand what you are trying to test here, could you clarify please?

Comment (nikhilgoel1997) [reply src/mlpack/tests/main_tests/adaboost_test.cpp]: If a model is given as input, and after that a different weak learner is specified (Maybe with the testing data), it should be ignored.
The same is written in adaboost_main.cpp line 132. I'm trying to test this. 
Is it wrong or have I misunderstood it in some way?

Comment (rcurtin) [reply src/mlpack/tests/main_tests/adaboost_test.cpp]: But if the weak learner is ignored, the program won't throw an exception (or at least it shouldn't), it should just print a warning message.  I don't think we have a good way to test if warning output is given, so I wouldn't worry about testing that.

**Comment by rcurtin** [comment]
You are right, I overlooked that you had already written a test to ensure that `weak_learner` is valid.  Sorry about the confusion!

**Comment by rcurtin** [comment]
Just FYI, this probably needs to be updated to handle the changes in #1214.  Here is an example of the type of changes that might need to be made:
https://github.com/mlpack/mlpack/pull/1214/files#diff-8fcb02fea0b4d1d0ee48e346e252a10b

**Comment by nikhilgoel1997** [comment]
Thank you for your help!
I've removed the weakLearnerIgnoredTest and added the three tests that was asked.
Is there anything else I need to add?

File: src/mlpack/tests/main_tests/adaboost_test.cpp [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+  int flag = 0;
+
+  for (size_t i = 0; i < output.n_elem; ++i)
+  {
+    if (output[i] != outputPerceptron[i])
+    {
+      flag = 1;
+      break;
+    }
+  }
```
Comment (rcurtin) [inline src/mlpack/tests/main_tests/adaboost_test.cpp]: You could also do

```
BOOST_REQUIRE_GT(arma::accu(output != outputPerceptron), 1);
```

**Comment by nikhilgoel1997** [comment]
Thanks for the cleaner way to write that test!
I've added my name to COPYRIGHT and core.hpp! It has been a great learning experience and I'll try my best to apply this experience in my future contributions to scilab! Thank you for guiding me!
Yes, I would love the stickers.
I was unable to run these tests/valgrind on my machine. The console gave the error as "No test case matching the filter". Am I missing something?

**Comment by nikhilgoel1997** [comment]
@rcurtin I don't understand why the Static Code Analysis check is failing. Is there something wrong with that build? Even the console output is not giving a good reason.

**Comment by zoq** [comment]
@mlpack-jenkins test this please

**Comment by zoq** [comment]
> If you like, you can check this test with valgrind:
> 
> valgrind --leak-check=full --track-origins=yes bin/mlpack_test -t AdaBoostMainTest

This check should now be covered by the Memory checks job.

**Comment by rcurtin** [comment]
Sweet!  I will stop checking manually then. :)

File: src/mlpack/tests/main_tests/adaboost_test.cpp [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+};
+
+BOOST_FIXTURE_TEST_SUITE(AdaBoostMainTest, AdaBoostTestFixture);
+
+/**
+ * Check that number of output labels and number of input
+ * points are equal.
+ */
+
```
Comment (zoq) [inline src/mlpack/tests/main_tests/adaboost_test.cpp]: Super pedantic style issue; remove the extra line.

**Comment by rcurtin** [comment]
With two approvals I will go ahead and merge this now.  Thank you again Nikhil for the nice tests. :)

**Pull request merged**